### ISSUE TITLE
Split gigantic ZIO.scala file into several ones

### ISSUE THREAD

**Issue by evis** [zio/zio#4200]
It's so hard to work with 4k LOC file. It's easy to get lost here.

For ZIO project developer, it would be more convenient to deal with smaller files with several functions, e.g.:

```scala
// file ZIOBrackets.scala
private[zio] trait ZIOBrackets[-R, +E, +A] {

  final def bracket[R1 <: R, E1 >: E, B](
    release: A => URIO[R1, Any],
    use: A => ZIO[R1, E1, B]
  ): ZIO[R1, E1, B] = ZIO.bracket(self, release, use)

  // other bracket functions here...
}

// file ZIO.scala
sealed trait ZIO[-R, +E, +A] extends ZIOBrackets[R, E, A] with ...
```

I'm not sure, is there any significant performance penalty, though.

**Comment by adamgfraser** [comment]
I'm 👎 on this. It is a large file but splitting things into multiple files has its own disadvantages for discoverability as people have to look in these separate files and understand the logic for what goes in each file. We did something similar with `Gen` in ZIO Test where we put some generators in other files with traits that we extended and I tend to think it has hurt discoverability. I think it is actually relatively straightforward to navigate despite its size just based on trait / companion object and then alphabetization. If anything I think there is a little more we could do to cleanup the syntax classes / partially applied classes / other helpers at the bottom of the file, though I don't think that is a huge problem.
